timemachine/fe/absolute_hydration.py

```diff
@@ -0,0 +1,142 @@
+"""Absolute hydration free energies"""
+
+from functools import partial
+
+import numpy as np
+
+from timemachine.constants import BOLTZ
+from timemachine.fe.lambda_schedule import construct_pre_optimized_absolute_lambda_schedule_solvent
+from timemachine.ff import Forcefield
+from timemachine.md import enhanced, moves
+from timemachine.md.smc import conditional_multinomial_resample
+from timemachine.md.states import CoordsVelBox
+from timemachine.utils import bind_potentials, construct_potential
+
+
+def generate_endstate_samples(num_samples, solvent_samples, ligand_samples, ligand_log_weights, num_ligand_atoms):
```

Comment:
nit: worth adding types here? e.g. is `solvent_samples: Sequence[CoordsVelBox]`?